cli/src/test/java/com/devonfw/tools/ide/log/IdeTestLoggerAssertion.java

```diff
@@ -89,23 +89,38 @@ private void hasEntries(boolean nothingElseInBetween, IdeLogEntry... exprectedEn
 
     assert (exprectedEntries.length > 0);
     int i = 0;
-    for (IdeLogEntry entry : exprectedEntries) {
-      if (entry.equals(exprectedEntries[i])) {
+    int max = 0;
+    for (IdeLogEntry entry : this.entries) {
+      if (exprectedEntries[i].matches(entry)) {
         i++;
       } else {
         if (nothingElseInBetween) {
           i = 0;
-        } else if (entry.equals(exprectedEntries[i])) {
+        } else if (exprectedEntries[0].matches(entry)) {
           i = 1;
         }
       }
       if (i == exprectedEntries.length) {
         return;
       }
+      if (i > max) {
+        max = i;
+      }
     }
     StringBuilder error = new StringBuilder(4096);
-    error.append("Could not find expected log entries:\n");
-    for (IdeLogEntry entry : exprectedEntries) {
+    if (max > 0) {
+      error.append("Found expected log entries:\n");
+      for (i = 0; i < max; i++) {
+        IdeLogEntry entry = exprectedEntries[i];
+        error.append(entry.level());
+        error.append(":");
+        error.append(entry.message());
+        error.append('\n');
+      }
+    }
+    error.append("\nBut could not find expected log entries:\n");
+    for (i = max; i < exprectedEntries.length; i++) {
+      IdeLogEntry entry = exprectedEntries[i];
```

Comment:
I think this might only work fully as intended for when `nothingElseInBetween` is set to true. 
Consider that some (faulty) code produces the following Info log:
```
INFO - SomeLogger - A
INFO - SomeLogger - C
```
And consider the test calls `assertThat(context).logAtInfo().hasEntries("A", "B", "C")` to check for all 3 messages to be present as intendedn. 
I would expect the test to fail and state the missing "B" message.

However currently the test will fail with the following info to the developer:
```
Found expected log entries:
INFO:A

But could not find expected log entries:
INFO:B
INFO:C
```
even though the "C" message had been present.

However, this does not affect the correctness of the test, only the information provided upon failure.

Reply:
This is just a matter of how we report the error in cases where the assertion is failing.
My intention was just to be helpful and show the first line that did not match so the developer can faster find the problem. This feature already helped me in various situations.
Your expectation would be extraordinary complex to implement. Also consider that I may expect `"A", "A", "B", "A"` and then what should I get reported if I however have `"A", "B", "B", "A"` in the log?
To me it is important that the assertion is properly verified. There error message should be helpful for the developer but must not be perfect. If you want to dig deeper here, you could argue this block
```
} else if (expectedEntries[0].matches(entry)) {
  i = 1;
```
is somehow not perfectly accurate for every edge-case but I think we can still fix it whenever this problem should hit us (IMHO not likely to happen in the next decade, but we will see).

Reply:
OK. I improved the error reporting to avoid confusion with a simple approach:
I only report the first exepcted log entry that is not matching.
Developer will see his test and all the expected log entries he is expecting so I will resolve this conversation.